VSCode/armsnippets.json

```diff
@@ -506,6 +506,123 @@
+        "                   \"action\": {",
+        "                      \"type\": \"${12:Snat,Dnat}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                      {",
+        "                          \"name\": \"${13:natRuleName}\",",
+        "                          \"description\": \"${14:natRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${15:listOfSourceAddresses}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"${16:listOfDestinationAddresses}\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${17:destinationPortRange:1-64000,*}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${18:TCP,UDP,Any,ICMP}\"",
+        "                          ],",
+        "                          \"translatedAddress\": \"${19:translatedIpAddress}\",",
+        "                          \"translatedPort\": \"${20:translatedIpPort}\"",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"networkRuleCollections\": [",
+        "           {",
+        "              \"name\": \"${21:networkCollectionName}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${22:priority:100-65000}\",",
+        "                   \"action\": {",
+        "                      \"type\": \"${23:Allow,Deny}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                      {",
+        "                          \"name\": \"${24:networkRuleName}\",",
+        "                          \"description\": \"${25:networkRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${29:listOfSourceAddresses}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"${26:listOfDestinationAddresses}\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${27:destinationPortRange:1-64000,*}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${28:TCP,UDP,Any,ICMP}\"",
+        "                          ]",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"ipConfigurations\": [",
```

Comment:
Does it make sense to include the publicIP and subnet resources in this snippet so you can use the resourceId() function to generate the right values for these "id" properties?

Reply:
Potentially. One of the issues I have had in general with the snippets is I frequently want the minimum allowed values to deploy. In the case of the FW I don't actually need the public IP resource (although it is useless without it). But, when building complex de-constructed templates I won't be passing those values from this section, but likely from existing resources, or resources that will be created in linked templates with output passed back.

Reply:
What about a concat instead referencing an existing resource? I think that makes sense.

Reply:
Or a call to resourceId() with tokens for resource group and resource name
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#resourceid

Reply:
Yup, thank you. Inbound update coming that should be good.